tests/unittests/ontology_validation_tests.py

```
@@ -0,0 +1,145 @@
+                    component="dropdown",
+                    when=[When(WhenOperator.EQUALS, field="b", value="x")],
+                ),
+                AttributeSpec(
+                    name="b",
+                    type="str",
+                    component="dropdown",
+                    when=[When(WhenOperator.EQUALS, field="a", value="y")],
+                ),
+            ],
+        )
+        with self.assertRaises(ValueError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)